src/main/java/org/opensearch/ad/rest/RestAnomalyDetectorJobAction.java

```diff
@@ -120,6 +116,11 @@ protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOExc
             rawPath
         );
 
+        // Ensure job details are registered with Job Scheduler prior to creating a job, no-op for historical request
+        if (rawPath.endsWith(RestHandlerUtils.START_JOB) && !registeredJobDetails && !request.hasContent()) {
```

Comment:
This works, but I'm wondering if we couldn't leverage the route handlers to do this first conditional.  

You could have a `handleStartRequest` and `handleStopRequest`; both delegating to this `prepareRequest()` method with an argument defining whether it's a start.

Reply:
This is a good idea, ill make this change